src/System.Device.Gpio/System/Device/Gpio/Drivers/Libgpiod/LibGpiodDriver.cs

```diff
@@ -1,177 +1,492 @@
     {
-        LibGpiodDriverFactory.VersionedLibgpiodDriver versionedLibgpiodDriver = LibGpiodDriverFactory.Instance.Create(gpioChip, driverVersion);
-        _driver = versionedLibgpiodDriver.LibGpiodDriver;
-        Version = versionedLibgpiodDriver.DriverVersion;
+        GPIOD_LINE_REQUEST_FLAG_OPEN_DRAIN = (1UL << 0),
+        GPIOD_LINE_REQUEST_FLAG_OPEN_SOURCE = (1UL << 1),
+        GPIOD_LINE_REQUEST_FLAG_ACTIVE_LOW = (1UL << 2),
+        GPIOD_LINE_REQUEST_FLAG_BIAS_DISABLE = (1UL << 3),
+        GPIOD_LINE_REQUEST_FLAG_BIAS_PULL_DOWN = (1UL << 4),
+        GPIOD_LINE_REQUEST_FLAG_BIAS_PULL_UP = (1UL << 5)
     }
 
     /// <summary>
-    /// Version of the libgpiod driver
+    /// Construct an instance
     /// </summary>
-    [Experimental(DiagnosticIds.SDGPIO0001, UrlFormat = DiagnosticIds.UrlFormat)]
-    public LibGpiodDriverVersion Version { get; protected set; }
+    /// <param name="gpioChip">Number of the gpio Chip. Default 0</param>
+    public LibGpiodDriver(int gpioChip = 0)
+    {
+        if (Environment.OSVersion.Platform != PlatformID.Unix)
+        {
+            throw new PlatformNotSupportedException($"{GetType().Name} is only supported on Linux/Unix.");
+        }
+
+        try
+        {
+            _pinNumberLock = new object();
+            _chipNumber = gpioChip;
+            _chip = LibgpiodV1.gpiod_chip_open_by_number(gpioChip);
+            if (_chip == null || _chip.IsInvalid || _chip.IsClosed)
+            {
+                throw ExceptionHelper.GetIOException(ExceptionResource.NoChipFound, Marshal.GetLastWin32Error());
+            }
+
+            _pinCount = LibgpiodV1.gpiod_chip_num_lines(_chip);
+            _pinNumberToEventHandler = new ConcurrentDictionary<int, LibGpiodV1DriverEventHandler>();
+            _pinNumberToSafeLineHandle = new ConcurrentDictionary<int, LineHandle>();
+            _pinValue = new ConcurrentDictionary<int, PinValue>();
+        }
+        catch (DllNotFoundException)
+        {
+            throw ExceptionHelper.GetPlatformNotSupportedException(ExceptionResource.LibGpiodNotInstalled);
+        }
+    }
 
     /// <summary>
-    /// A collection of driver versions that correspond to the installed versions of libgpiod on this system. Each driver is dependent
-    /// on specific libgpiod version/s. If the collection is empty, it indicates that libgpiod might not be installed or could not be detected.
+    /// Returns the set of available chips for this driver
     /// </summary>
-    [Experimental(DiagnosticIds.SDGPIO0001, UrlFormat = DiagnosticIds.UrlFormat)]
-    public static LibGpiodDriverVersion[] GetAvailableVersions() => LibGpiodDriverFactory.Instance.DriverCandidates;
+    /// <returns>A list of <see cref="GpioChipInfo"/> instances</returns>
+    public static IList<GpioChipInfo> GetAvailableChips()
+    {
+        List<GpioChipInfo> result = new List<GpioChipInfo>();
+        var iterator = new SafeChipIteratorHandle(LibgpiodV1.gpiod_chip_iter_new());
```

Comment:
Should gpiod_chip_iter_new return SafeHandle?

Reply:
I prever it explicitly. See the confusion above.